src/java.base/share/classes/sun/security/ssl/SunX509KeyManagerImpl.java

```diff
@@ -251,12 +251,28 @@ public String chooseEngineClientAlias(String[] keyType,
     @Override
     public String chooseServerAlias(String keyType,
             Principal[] issuers, Socket socket) {
-        /*
-         * We currently don't do anything with socket, but
-         * someday we might.  It might be a useful hint for
-         * selecting one of the aliases we get back from
-         * getServerAliases().
-         */
+        return chooseServerAlias(
+                keyType, issuers, getAlgorithmConstraints(socket));
+    }
+
+    /*
+     * Choose an alias to authenticate the server side of an
+     * <code>SSLEngine</code> connection given the public key type
+     * and the list of certificate issuer authorities recognized by
+     * the peer (if any).
+     *
+     * @since 1.5
```

Comment:
I would remove the `@since 1.5` from these methods. It isn't relevant anymore since this is an internal class and that version is no longer supported. That version info is in the `X509ExtendedKeyManager` API which is sufficient.